src/PlusDataCollection/WinProbe/vtkPlusWinProbeVideoSource.cxx

```diff
@@ -16,526 +16,700 @@ See License.txt for details.
-    deviceConfig->SetUnsignedLongAttribute("MinValue", this->GetMinValue());
-    deviceConfig->SetUnsignedLongAttribute("MaxValue", this->GetMaxValue());
-    deviceConfig->SetUnsignedLongAttribute("LogLinearKnee", this->GetLogLinearKnee());
-    deviceConfig->SetUnsignedLongAttribute("LogMax", this->GetLogMax());
-
-    deviceConfig->SetVectorAttribute("TimeGainCompensation", 8, m_timeGainCompensation);
-    deviceConfig->SetVectorAttribute("FocalPointDepth", 4, m_focalPointDepth);
-
-    return PLUS_SUCCESS;
+  XML_FIND_DEVICE_ELEMENT_REQUIRED_FOR_WRITING(deviceConfig, rootConfigElement);
+
+  deviceConfig->SetAttribute("TransducerID", this->m_transducerID.c_str());
+  deviceConfig->SetAttribute("UseDeviceFrameReconstruction", this->m_UseDeviceFrameReconstruction ? "TRUE" : "FALSE");
+  deviceConfig->SetAttribute("SpatialCompoundEnabled", this->GetSpatialCompoundEnabled() ? "TRUE" : "FALSE");
+  deviceConfig->SetFloatAttribute("TxTxFrequency", this->GetTxTxFrequency());
+  deviceConfig->SetFloatAttribute("SSDepth", this->GetSSDepth());
+  deviceConfig->SetFloatAttribute("SpatialCompoundAngle", this->GetSpatialCompoundAngle());
+  deviceConfig->SetIntAttribute("SpatialCompoundCount", this->GetSpatialCompoundCount());
+  deviceConfig->SetUnsignedLongAttribute("Voltage", this->GetVoltage());
+  deviceConfig->SetUnsignedLongAttribute("MinValue", this->GetMinValue());
+  deviceConfig->SetUnsignedLongAttribute("MaxValue", this->GetMaxValue());
+  deviceConfig->SetUnsignedLongAttribute("LogLinearKnee", this->GetLogLinearKnee());
+  deviceConfig->SetUnsignedLongAttribute("LogMax", this->GetLogMax());
+
+  deviceConfig->SetVectorAttribute("TimeGainCompensation", 8, m_timeGainCompensation);
+  deviceConfig->SetVectorAttribute("FocalPointDepth", 4, m_focalPointDepth);
+
+  return PLUS_SUCCESS;
 }
 
 // ----------------------------------------------------------------------------
-vtkPlusWinProbeVideoSource *thisPtr = NULL;
+vtkPlusWinProbeVideoSource* thisPtr = nullptr;
 
-//----------------------------------------------------------------------------
+//-----------------------------------------------------------------------------
 // This callback function is invoked after each frame is ready
-int __stdcall frameCallback(int length, char * data, char *hHeader, char *hGeometry)
+int __stdcall frameCallback(int length, char* data, char* hHeader, char* hGeometry)
+{
+  thisPtr->FrameCallback(length, data, hHeader, hGeometry);
+  return length;
+}
+
+//-----------------------------------------------------------------------------
+void vtkPlusWinProbeVideoSource::ReconstructFrame(char* data)
 {
-    thisPtr->FrameCallback(length, data, hHeader, hGeometry);
-    return length;
+  uint16_t* frame = reinterpret_cast<uint16_t*>(data + 16);
+  assert(m_BModeBuffer.size() == m_samplesPerLine * m_transducerCount);
+  const float logFactor = m_OutputKnee / std::log(1 + m_Knee);
+
+  #pragma omp parallel for
```

Comment:
I don't see any OpenMP checking in the CMake. Will this compile if openmp is not available?

Reply:
Unrecognized `#pragma`s should be ignored by the compiler. See https://stackoverflow.com/a/17511382/276168